ci/set_github_status.sh

```diff
@@ -17,12 +19,14 @@ commit_status=${1}
 submit "${commit_status}" "$CI_COMMIT_SHA"
 
 # For Bors: get the latest commit before the merge to set the status.
+echo $CI_COMMIT_REF_NAME
 if [[ $CI_COMMIT_REF_NAME =~ ^(trying|staging)$ ]]; then
     parent_sha=`git rev-parse --verify -q "$CI_COMMIT_SHA"^2`
+    echo "git rev-parse --verify -q $CI_COMMIT_SHA^2 -> $?"
     if [[ $? -eq 0 ]]; then
```

Comment:
What if the exit code is not equal to zero, then everything is just fine? Isn't an else clause missing, throwing an error?

Reply:
Note that I did not originally write that script, only adapt it. I am not even sure why the exit code there can be 0.

Reply:
Not sure I understand you correctly, but if `git` runs successfully it will return 0 to signal that. If not you should raise an error. Additionally you introduced a bug here, echo will most likely always return 0 and you are now checking for its exit code instead of the one from `git`.

Reply:
The echo has already been removed

Reply:
The new version uses `set -e` to error out as soon as any command returns nonzero, from the log it will be apparent which command that was.